javascript/code-challenges/challenges-01.test.js

```diff
@@ -0,0 +1,211 @@
+'use strict';
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1
+
+Write a function named `addOne` that takes an array of numbers, and returns a new array of the numbers, incremented by 1.
+
+Use `forEach` to loop over the input array and work with each value.  Push the new value into a local array. Return the local array;
+------------------------------------------------------------------------------------------------ */
+
+const addOne = (arr) => {
+  const newArray = [];
+  // Solution code here...
+
+  arr.forEach(num => {
+    newArray.push(num + 1);
+  });
+
+  return newArray;
+
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function named `addExclamation` that takes an array of strings, and returns a new array of the same strings with an "!" added to the end.
+
+Use `forEach` to loop over the input array. Modify each string, and add the updated value into a local array. Return the local array;
+------------------------------------------------------------------------------------------------ */
+
+const addExclamation = (arr) => {
+  const newArray = []
+  // Solution code here...
+
+  arr.forEach(greet => {
+    newArray.push(`${greet}!`);
+  })
+
+  return newArray;
+
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function named `allUpperCase` that takes an array of strings, and returns a new array of the strings converted to upper case.
+
+Use `forEach` to loop over the input array. The modified strings should each be added into a local array. Return that local array.
+------------------------------------------------------------------------------------------------ */
+
+const allUpperCase = (arr) => {
+  const newArray = []
+  // Solution code here...
+
+  arr.forEach(caps => {
+    newArray.push(`${caps.toUpperCase()}`)
+  })
+
+  return newArray;
+
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+Write a function named `greeting` that takes in a single string and returns the string in all uppercase letters, and followed by an "!".
+
+Then, write a function named `speaker` that takes in an array of strings and a callback function. 
+
+Use `forEach` to build a new array of strings, each string modified by the callback. Return the new array. 
+------------------------------------------------------------------------------------------------ */
+
+const greeting = (word) => `${word.toUpperCase()}!`
+  // Solution code here...
+
+ 
+const speaker = (words, callback) => {
```

Comment:
Hey there, nice work so far! 👍 
 You are getting closer, take  another look at what you are adding in front of the forEach and in front of `push().
 
   In other words, what do you need to iterate over?  Is it speaker or something else perhaps?

For push, what data structure to you want to push results into?